client/my-sites/site-settings/form-general.jsx

```diff
@@ -309,7 +316,7 @@ class SiteSettingsFormGeneral extends Component {
 
 		return (
 			<div className={ classNames( classes ) }>
-				{ site && <QuerySiteSettings siteId={ site.ID } /> }
+				{ site && <QuerySiteSettings siteId={ this.props.siteId } /> }
```

Comment:
Why not keep `site.ID` in here and use it in all other places? Is there any particular reason? We should stick to one of the following options in my opinion:

```js
{ this.props.siteId  && <QuerySiteSettings siteId={ this.props.siteId } /> }
// or
{ site && <QuerySiteSettings siteId={ site.ID } /> }
```

If we pick the first one, then we can use existing destructuring:
```js
		const {
			handleSubmitForm,
			isRequestingSettings,
			isSavingSettings,
			site,
            siteId,
			siteIsJetpack,
			translate
		} = this.props;
```